src/store/users.js

```diff
@@ -1,101 +1,125 @@
-          user: newUser,
-        });
-        commit("setUser", { id: result.id, user: result });
-        return result.id;
-      } catch (error) {
-        commit("addError", error, { root: true });
-        return false;
-      }
-    },
-    async update({ commit, rootState }, { id, newUser }) {
-      try {
-        const result = await api.users.update(rootState.auth.apiToken, id, {
-          user: newUser,
-        });
-        commit("setUser", { id, user: result });
-        return true;
-      } catch (error) {
-        commit("addError", error, { root: true });
-        return false;
-      }
-    },
-    async destroy({ commit, rootState }, id) {
-      try {
-        await api.users.destroy(rootState.auth.apiToken, id);
-        commit("removeUser", id);
-        return true;
-      } catch (error) {
-        commit("addError", error, { root: true });
-        return false;
+export const useUsersStore = defineStore("users", () => {
+  const authStore = useAuthStore();
+  const errorsStore = useErrorsStore();
+
+  const users = useLocalStorage(
+    "users.users",
+    {},
+    { serializer: ShallowObjectSerializer },
+  );
+  const startLoading = ref(new Date(0));
+
+  function setUsers(payload) {
+    const oldUsers = users.value;
+    users.value = {};
+    for (let id in oldUsers) {
+      users.value[id] = oldUsers[id];
+    }
+    const loaded = new Date();
+    for (let user of payload) {
+      user.loaded = loaded;
+      users.value[user.id] = markRaw(user);
+    }
+  }
+
+  function setUser({ id, user }) {
+    const oldUsers = users.value;
+    users.value = {};
+    for (let id in oldUsers) {
+      users.value[id] = oldUsers[id];
+    }
+    user.loaded = new Date();
+    users.value[id] = markRaw(user);
+  }
+
+  function setStartLoading() {
+    startLoading.value = new Date();
+  }
+
+  function removeUser(id) {
+    del(users.value, id);
+  }
+
+  function removeOld() {
+    const oldUsers = users.value;
+    users.value = {};
+    for (let id in oldUsers) {
+      if (oldUsers[id].loaded > startLoading.value) {
+        users.value[id] = oldUsers[id];
       }
-    },
-  },
-  getters: {
-    users: (state) => Object.values(state.users),
-    usersByName: (state, getters) =>
-      getters.users.sort((u1, u2) => compareStrings(u1.name, u2.name)),
-  },
-};
+    }
+  }
+
+  async function index() {
+    const generator = api.users.index(authStore.apiToken);
+    try {
+      await fetchAllPinia(generator, setUsers, setStartLoading, removeOld);
+      return true;
+    } catch (error) {
+      errorsStore.addError(error);
+      return false;
+    }
+  }
+
+  async function create(newUser) {
+    try {
+      const result = await api.users.create(authStore.apiToken, {
+        user: newUser,
+      });
+      setUser({ id: result.id, user: result });
+      return result.id;
+    } catch (error) {
+      errorsStore.addError(error);
+      return false;
+    }
+  }
+
+  async function update(id, newUser) {
```

Comment:
Note that this API changed. Because pinia is a lot more flexible than vuex we can pass multiple arguments, which is the nice way to do this.